### PR TITLE
Changed astexport to just return the AST as a dictionary

### LINKED ISSUE #21: Divide the astexport.py module between pydetector and the driver
In the meeting planning we decided to split the functionality in the current `pydetector.astexport.py` module between the retrieval of the native AST data structure unmodified (but for the right Python version) in `pydetector` and the visitor + noop extractor + position updater in `python-driver`, reusing the data returned from `pydetector` to avoid doing a double parsing.

### PR DESCRIPTION
**Pull request by juanjux**
This implements the PyDetector part of the split. It will just take the native AST and convert it to a dictionary, without any further processing or modification of the information.

Why is doesn't just return directly what `ast.parse()` returns?

I've found that the object tree that `ast.parse()` returns is created on the `_ast` binary module and its internal binary format changes between version. When the AST is extracted using Python 2, since the driver runs Python 3, the unpickled object would be incompatible (it could also potentially happen between minor versions since compatibility of the AST is not warranted by the developers). So I just convert it to a dictionary with dictionary keys, values, lists et all that can be exported between versions without any problem.

The Python's driver part will follow and the integration tests could be checked there.

Fixes #21

### CODE REVIEW

File: pydetector/ast_checks.py [hunk 1]
```diff
@@ -85,6 +85,7 @@ def check_ast(code, try_other_on_sucess=False, verbosity=0,
                     print('>>>> ASTCHECK: error while parsing AST with Python%d:\n%s\n<<<< error output end'
                           % (PYMAJOR_OTHER, err))
         except:
+            print(format_exc())  # XXX
```
Comment (abeaumont) [inline pydetector/ast_checks.py]: Is this a left over?

File: pydetector/ast2dict.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+def ast2dict(codestr):
+    """ Returns the AST as a Python dictionary """
+    visitor = DictExportVisitor(codestr)
+    return visitor.parse()
+
+
+class DictExportVisitor(object):
+
+    def __init__(self, codestr, ast_parser=ast.parse):
+        self.codestr    = codestr
```
Comment (abeaumont) [inline pydetector/ast2dict.py]: Extra spaces

File: pydetector/ast2dict.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    def _nodedict(self, node, newdict, ast_type=None):
+        """
+        Shortcut that adds ast_type (if not specified),
+        lineno and col_offset to the node-derived dictionary
+        """
```
Comment (abeaumont) [inline pydetector/ast2dict.py]: These docstrings don't match with PEP8/257 requirements.

File: pydetector/ast2dict.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        content = \
+'''if True:
+    if False:
+        a = 6 + 7
+    else:
+        print('LAST')
+
+'''
+
+    print(export_dict(content))
```
Comment (abeaumont) [inline pydetector/ast2dict.py]: `export_dict` should be renamed to `ast2dict`

File: pydetector/ast2dict.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        newdict["_attributes"] = getattr(node, "_attributes", [])
+        return newdict
+
+    def parse(self):
+        tree = ast.parse(self.codestr, mode='exec')
+        res = self.visit(tree, root=True)
+        return res
+
+    def visit(self, node, root=False):
+        if isinstance(node, string_types) or\
```
Comment (abeaumont) [inline pydetector/ast2dict.py]: space after `or`?

File: pydetector/ast2dict.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+                (not isinstance(node, Sequence) and
+                 not isinstance(node, ast.AST)):
+            return node
+
+        nodedict = self._nodedict(node, {}, ast_type = node.__class__.__name__)
```
Comment (abeaumont) [inline pydetector/ast2dict.py]: no spaces around `=`

File: pydetector/ast2dict.py [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+    def visit_field(self, node):
+        if isinstance(node, ast.AST):
+            return self.visit(node)
+        elif isinstance(node, list) or isinstance(node, tuple):
+            return [self.visit(x) for x in node]
+        else:
+            return node
+
+    def visit_str(self, node):
```
Comment (abeaumont) [inline pydetector/ast2dict.py]: This method doesn't seem to be called.

**Comment by juanjux** [comment]
@abeaumont thank for the review; I fixed all those (also changed that docstring to a smaller comment since the method is private).

**Comment by abeaumont** [comment]
The docstring comment was a general one, since no docstring seems to be following it. But this doesn't seem something new, so feel free to merge, just a remainder for future improvements.

**Pull request merged**